src/Service/CartService.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+namespace Makaira\OxidConnectEssential\Service;
+
+use Makaira\OxidConnectEssential\Exception\InvalidCartItem;
+use OxidEsales\Eshop\Application\Model\Article;
+use OxidEsales\Eshop\Application\Model\Basket;
+use OxidEsales\Eshop\Application\Model\BasketItem;
+use OxidEsales\Eshop\Core\Exception\ArticleInputException;
+use OxidEsales\Eshop\Core\Exception\NoArticleException;
+use OxidEsales\Eshop\Core\Exception\OutOfStockException;
+use OxidEsales\Eshop\Core\Registry;
+use OxidEsales\Eshop\Core\Session;
+
+class CartService
+{
+    private Session $session;
+    private Basket $basket;
+
+    public function __construct()
+    {
+        $this->session = Registry::getSession();
+        $this->basket = $this->session->getBasket();
+    }
```

Comment:
Do not use `OxidEsales\Eshop\Core\Registry` inside our custom classes! Instead inject the dependencies.

```suggestion
    public function __construct(private Session $session, private Basket $basket)
    {
    }
```